Add some elliptic AMR projectors

nilsvu opened the pull request:
## Proposed changes

<!--
At a high level, describe what this PR does.
-->

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsvu committed:
Add AMR projector for elliptic sources

nilsvu committed:
Add AMR projector for elliptic analytic solutions

nilsvu committed:
Add AMR projector for elliptic DG operator

nilsvu committed:
Ignore a clang-tidy check

nilsvu committed:
Add AMR projector for elliptic domain

nilsvu committed:
Add AMR projector for elliptic faces & mortars

nilsvu committed:
Add AMR projector for elliptic background

kidder commented on .clang-tidy:
```diff
@@ -76,6 +76,8 @@ Checks: '*,
 -readability-magic-numbers,
 # Same as llvm-qualified-auto above.,
 -readability-qualified-auto,
+# Access specifies can be useful to structure code,
```
specifies -> specifiers

nilsvu commented:
@kidder I made this PR very small now, and will open follow-ups later. Could you review this please?

kidder reviewed (changes requested):
you should test the projectors

nilsvu committed:
Add AMR logging tag to component

Before it wasn't reliably added.

nilsvu committed:
Add test for elliptic AMR projectors

nilsvu commented:
@kidder Updated and added a test for p-AMR in action. h-AMR isn't supported by the action-testing framework yet as we discussed, so I can enable that once we have added support.

Please ignore the first two commits (cherry-picked from #5163 and #5781).

kidder reviewed (dismissed):
needs a rebase

nilsvu commented:
Rebased. Thanks for reviewing.
